### PR TITLE
Implemented Onboarding Real Real High Fidelity

### PR DESCRIPTION
**Pull request by rs929**
## Overview

- Began implementation of REAL REAL HIGH FIDELITY for Scooped Onboarding Screens
- Began implementation of scooped onboarding animation (WIP)
- Delegated Trip Details information to TripDetailsViewController
- Added SDWebImage to get images from URL
- Pod update

## Changes Made

- Implemented all designs for Scooped Onboarding screens
- Started implementation of Scooped onboarding animation but still W.I.P
- Delegated information from matchesViewController so that TripDetailsViewController now displays information about the ride that is selected from the matches
- Pod install SDWebImage to get image from URL

## Next Steps (delete if not applicable)

- Fully implement Scooped onboarding animation
- Finish conversion to RRHF
- Implement networking for updating user preferences
- Finalize onboarding RRHF

## Screenshots (delete if not applicable)

Simulator Screen Shot - iPhone 14 Pro - 2023-03-16 at 20 42 28
Simulator Screen Shot - iPhone 14 Pro - 2023-03-16 at 20 42 32
Simulator Screen Shot - iPhone 14 Pro - 2023-03-16 at 20 42 41
Simulator Screen Shot - iPhone 14 Pro - 2023-03-16 at 20 42 45
Simulator Screen Shot - iPhone 14 Pro - 2023-03-16 at 20 42 52

### CODE REVIEW

File: Scoop/Controller/Onboarding/FavoritesViewController.swift [hunk 1]
```diff
@@ -61,60 +66,117 @@ class FavoritesViewController: OnboardingViewController {
...
-        stackView.setCustomSpacing(spacing, after: snackLabel)
-        
         snackTextField.textColor = .darkGray
-        snackTextField.placeholder = "Enter snack..."
+        snackTextField.delegate = self
+        snackTextField.attributedPlaceholder = NSAttributedString(
+            string: "Roadtrip snack",
+            attributes: [NSAttributedString.Key.foregroundColor: UIColor.offBlack])
         stackView.addArrangedSubview(snackTextField)
         snackTextField.snp.makeConstraints { make in
```
Comment (tiffany-pann) [inline Scoop/Controller/Onboarding/FavoritesViewController.swift:76]: Linebreak here

File: Scoop/Controller/Onboarding/FavoritesViewController.swift [hunk 2]
```diff
@@ -61,60 +66,117 @@ class FavoritesViewController: OnboardingViewController {
...
-        stopLabel.textColor = .black
-        stackView.addArrangedSubview(stopLabel)
-        stackView.setCustomSpacing(spacing, after: stopLabel)
+        songTextField.textColor = .darkGray
+        songTextField.delegate = self
+        songTextField.attributedPlaceholder = NSAttributedString(
+            string: "Roadtrip song",
+            attributes: [NSAttributedString.Key.foregroundColor: UIColor.offBlack])
+        stackView.addArrangedSubview(songTextField)
+        songTextField.snp.makeConstraints { make in
```
Comment (tiffany-pann) [inline Scoop/Controller/Onboarding/FavoritesViewController.swift:88]: Linebreak here

File: Scoop/Controller/Onboarding/FavoritesViewController.swift [hunk 3]
```diff
@@ -61,60 +66,117 @@ class FavoritesViewController: OnboardingViewController {
...
+        }
         
         stopTextField.textColor = .darkGray
-        stopTextField.placeholder = "Enter stop..."
+        stopTextField.delegate = self
+        stopTextField.attributedPlaceholder = NSAttributedString(
+            string: "Roadtrip stop",
+            attributes: [NSAttributedString.Key.foregroundColor: UIColor.offBlack])
         stackView.addArrangedSubview(stopTextField)
         stopTextField.snp.makeConstraints { make in
```
Comment (tiffany-pann) [inline Scoop/Controller/Onboarding/FavoritesViewController.swift:100]: Linebreak here too, and in the other subview setups

File: Scoop/Controller/Onboarding/OnboardingViewController.swift [hunk 4]
```diff
@@ -25,27 +25,43 @@ class OnboardingViewController: UIViewController {
...
-        button.setImage(UIImage(named: "nextbutton"), for: .normal)
-        button.clipsToBounds = true
+        button.setTitle("Next", for: .normal)
+        button.backgroundColor = .secondaryGreen
+        button.layer.cornerRadius = 25
+        button.setTitleColor(.offBlack, for: .normal)
+        button.titleLabel?.font = .systemFont(ofSize: 16, weight: .bold)
         button.addAction(action, for: .touchUpInside)
         view.addSubview(button)
         button.snp.makeConstraints { make in
```
Comment (tiffany-pann) [inline Scoop/Controller/Onboarding/OnboardingViewController.swift:44]: Line break

File: Scoop/Controller/Onboarding/OnboardingViewController.swift [hunk 5]
```diff
@@ -25,27 +25,43 @@ class OnboardingViewController: UIViewController {
...
+        backButton.layer.borderColor = UIColor.offBlack.cgColor
+        backButton.layer.borderWidth = 1
+        backButton.backgroundColor = .white
+        backButton.layer.cornerRadius = 25
+        backButton.setTitleColor(.scoopDarkGreen, for: .normal)
+        backButton.titleLabel?.font = .systemFont(ofSize: 16, weight: .bold)
         backButton.addTarget(self, action: #selector(prevVC), for: .touchUpInside)
-        self.navigationController?.navigationBar.addSubview(backButton)
+        view.addSubview(backButton)
         backButton.snp.makeConstraints { make in
```
Comment (tiffany-pann) [inline Scoop/Controller/Onboarding/OnboardingViewController.swift:70]: Line break here too

**Review by tiffany-pann — APPROVED** [review]
Looks good! Great work Richie 🚀

File: Scoop/Controller/Onboarding/AboutYouViewController.swift [hunk 6]
```diff
@@ -55,14 +61,18 @@ class AboutYouViewController: OnboardingViewController {
...
         let spacing = 12.0
         let screenSize = UIScreen.main.bounds
-        let labelFont = UIFont(name: "Rambla-Regular", size: 16)
         let textFieldBorderWidth = 1.0
-        let textFieldCornerRadius = 8.0
+        let textFieldCornerRadius = 4.0
         let textFieldFont = UIFont(name: "SFPro", size: 16)
+        let textFieldHeight = 56
+        
+        if screenSize.height < 2000 {
```
Comment (ReadeP) [inline Scoop/Controller/Onboarding/AboutYouViewController.swift:73]: This feels weird? Are you sure we are going about laying out our UI correctly? What's the justification for this?

File: Scoop/Controller/Onboarding/AboutYouViewController.swift [hunk 7]
```diff
@@ -55,14 +61,18 @@ class AboutYouViewController: OnboardingViewController {
...
+            stackviewMultiplier = 0.15
```
Comment (ReadeP) [inline Scoop/Controller/Onboarding/AboutYouViewController.swift:74]: `stackViewMultiplier`

File: Scoop/Controller/Onboarding/AboutYouViewController.swift [hunk 8]
```diff
@@ -75,96 +85,164 @@ class AboutYouViewController: OnboardingViewController {
...
+    }
+
 }
 
 // MARK: - UITextFielDelegate
 extension AboutYouViewController: UITextFieldDelegate {
     
     func textField(_ textField: UITextField, shouldChangeCharactersIn range: NSRange, replacementString string: String) -> Bool {
-        return false
+        if textField == yearTextField || textField == pronounsTextField {
```
Comment (ReadeP) [inline Scoop/Controller/Onboarding/AboutYouViewController.swift:204]: return `!(textField == yearTextField || textField == pronounsTextField)`

File: Scoop/Controller/Onboarding/AboutYouViewController.swift [hunk 9]
```diff
@@ -75,96 +85,164 @@ class AboutYouViewController: OnboardingViewController {
...
+            pronounsLabel.isHidden = false
+        } else if textField == hometownTextField {
+            hometownLabel.textColor = .scoopDarkGreen
+            hometownLabel.isHidden = false
+        } else {
+            yearLabel.textColor = .scoopDarkGreen
+            yearLabel.isHidden = false
+        }
+        
+        return
```
Comment (ReadeP) [inline Scoop/Controller/Onboarding/AboutYouViewController.swift:229]: delete `return`

File: Scoop/Controller/Onboarding/AboutYouViewController.swift [hunk 10]
```diff
@@ -75,96 +85,164 @@ class AboutYouViewController: OnboardingViewController {
...
+            nameLabel.textColor = .textFieldBorderColor
+        } else if textField == pronounsTextField {
+            pronounsLabel.textColor = .textFieldBorderColor
+        } else if textField == hometownTextField {
+            hometownLabel.textColor = .textFieldBorderColor
+        } else {
+            yearLabel.textColor = .textFieldBorderColor
+        }
+        
+        return
```
Comment (ReadeP) [inline Scoop/Controller/Onboarding/AboutYouViewController.swift:245]: here too

File: Scoop/Controller/Onboarding/FavoritesViewController.swift [hunk 11]
```diff
@@ -61,60 +66,120 @@ class FavoritesViewController: OnboardingViewController {
...
+            snackLabel.isHidden = false
+        } else if textField == songTextField {
+            songLabel.textColor = .scoopDarkGreen
+            songLabel.isHidden = false
+        } else {
+            stopLabel.textColor = .scoopDarkGreen
+            stopLabel.isHidden = false
+        }
+        
+        return
```
Comment (ReadeP) [inline Scoop/Controller/Onboarding/FavoritesViewController.swift:169]: delete return

File: Scoop/Controller/Onboarding/FavoritesViewController.swift [hunk 12]
```diff
@@ -61,60 +66,120 @@ class FavoritesViewController: OnboardingViewController {
...
+        textField.layer.borderColor = UIColor.textFieldBorderColor.cgColor
+        if textField == snackTextField {
+            snackLabel.textColor = .textFieldBorderColor
+        } else if textField == songTextField {
+            songLabel.textColor = .textFieldBorderColor
+        } else {
+            stopLabel.textColor = .textFieldBorderColor
+        }
+        
+        return
```
Comment (ReadeP) [inline Scoop/Controller/Onboarding/FavoritesViewController.swift:183]: delete return

File: Scoop/Controller/Onboarding/OnboardingPageViewController.swift [hunk 13]
```diff
@@ -126,3 +139,7 @@ extension OnboardingPageViewController: OnboardingDelegate {
         presentNextVC(nextVC)
     }
 }
+
+protocol AnimationDelegate: UIViewController {
+  func animateCar(startPage: Int, endPage: Int)
```
Comment (ReadeP) [inline Scoop/Controller/Onboarding/OnboardingPageViewController.swift:144]: indenting looks off here

**Review by ReadeP — APPROVED** [review]

**Pull request merged**